Loop/Views/InsulinModelSelection.swift

```diff
@@ -0,0 +1,317 @@
+//
+//  InsulinModelSelection.swift
+//  Loop
+//
+//  Created by Michael Pangburn on 7/14/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import HealthKit
+import SwiftUI
+import LoopCore
+import LoopKit
+import LoopKitUI
+import LoopUI
+
+
+final class InsulinModelSelectionViewModel: ObservableObject {
+    @Published var insulinModelSettings: InsulinModelSettings
+    var insulinSensitivitySchedule: InsulinSensitivitySchedule
+
+    static let defaultInsulinSensitivitySchedule = InsulinSensitivitySchedule(unit: .milligramsPerDeciliter, dailyItems: [RepeatingScheduleValue<Double>(startTime: 0, value: 40)])!
+
+    static let defaultWalshInsulinModelDuration = TimeInterval(hours: 6)
+    static let validWalshModelDurationRange = TimeInterval(hours: 2)...TimeInterval(hours: 8)
+
+    var walshActionDuration: TimeInterval {
+        get {
+            if case .walsh(let walshModel) = insulinModelSettings {
+                return walshModel.actionDuration
+            } else {
+                return Self.defaultWalshInsulinModelDuration
+            }
+        }
+        set {
+            precondition(Self.validWalshModelDurationRange.contains(newValue))
+            insulinModelSettings = .walsh(WalshInsulinModel(actionDuration: newValue))
+        }
+    }
+
+    init(insulinModelSettings: InsulinModelSettings, insulinSensitivitySchedule: InsulinSensitivitySchedule?) {
+        self._insulinModelSettings = Published(wrappedValue: insulinModelSettings)
+        self.insulinSensitivitySchedule = insulinSensitivitySchedule ?? Self.defaultInsulinSensitivitySchedule
+    }
+}
+
+struct InsulinModelSelection: View, HorizontalSizeClassOverride {
+    struct FeatureFlags {
```

Comment:
[nit] consider another name like `EnabledModels` or something like that, just because `FeatureFlags` has a more global meaning.